### PR TITLE
Describe middleware

### PR DESCRIPTION
**Pull request by EugeniyBurmistrov**

### CODE REVIEW

**Review by const314 — APPROVED** [review]

File: docs/api/middleware.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+### middlewareContext
+
+A command middleware handler's `middlewareContext` argument is an object with the following fields:
+
+| Field Name | Description                                                     |
+| ---------- | --------------------------------------------------------------- |
+| req        | Stores data that describes the currently processed HTTP request |
+| res        | Contains function used to configure the server's response       |
+
+Both `req` and `res` fields are included only if the command arrives from a client. If the command is generated on the server (for example, by a saga or API handler), these fields are omitted.
```
[the remark this answers is not in the document]
Comment (EugeniyBurmistrov) [reply docs/api/middleware.md:42]: Can I change "returns" to "sends"? I think, it sounnds more technically correct.

[the remark this answers is not in the document]
Comment (albertov05) [reply docs/api/middleware.md:42]: Okay, sounds good.

```suggestion
Both `req` and `res` fields are included only if the client sends the command. If the command is generated on the server (for example, by a saga or API handler), these fields are omitted.
```

File: docs/middleware.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
+id: middleware
+title: Middleware
+description: This describes how to use middleware in aggregates, read model resolvers and projections.
```
[the remark this answers is not in the document]
Comment (EugeniyBurmistrov) [reply docs/middleware.md:4]: Yes, I missed the word "topic". Added the word and comma.

File: docs/application-configuration.md [hunk 3]
```diff
@@ -343,6 +343,44 @@ const prodConfig = {
 }
 ```
```
[the remark this answers is not in the document]
Comment (EugeniyBurmistrov) [reply docs/application-configuration.md:345]: Done

File: docs/application-configuration.md [hunk 4]
```diff
@@ -343,6 +343,44 @@ const prodConfig = {
...
+    readModel: {
+        projection: ['common/middlewares/projection-middleware.js'],
+        resolver: ['common/middlewares/resolver-middleware.js']
+    }
+  },
+  ...
+}
+```
+
 ### jwtCookie
```
[the remark this answers is not in the document]
Comment (EugeniyBurmistrov) [reply docs/application-configuration.md:385]: Done
>Please rewrite without using "By default".

Changed to:
"If this option is omitted, the package name defined in the application's package.json file is reused as the application's name."

File: docs/middleware.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (albertov05) [reply docs/middleware.md]: Thank you.

File: docs/application-configuration.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (albertov05) [reply docs/application-configuration.md:345]: Thanks

File: docs/application-configuration.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (albertov05) [reply docs/application-configuration.md:385]: Sounds good. Thank you!

**Pull request merged**